app/routers/accounts.py

```diff
@@ -189,15 +196,40 @@ async def update_account(
     )
 
 
-@router.get("/{id}/users", response_model=LimitOffsetPage[AccountUserRead])
+@router.get("/{id}/users", response_model=LimitOffsetPage[UserRead])
 async def list_account_users(
-    account: Annotated[Account, Depends(fetch_account_or_404)], auth_context: CurrentAuthContext
+    account: Annotated[Account, Depends(fetch_account_or_404)],
+    auth_context: CurrentAuthContext,
+    user_repo: UserRepository,
 ):
     """
-    if auth_context.account.type == AFFILIATE && auth_context.account != account
-        403
+    This Endpoint lists all the users bound to a given account id.
+    The output is paginated by default.
+    Raises:
+        - HTTPException with status 400 if the given account is different from the context account
+        - HTTPException 404 if the provided account's id doesn't exist.
+    Returns a list of accounts if any.
     """
-    pass
+
+    if auth_context.account.type == AccountType.AFFILIATE and auth_context.account != account:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="Cheating is bad. Don't do it.",
```

Comment:
For a 403 we can use a funny message but for a 400 we should use something meaningful. Sorry to not see it before.